mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -1784,11 +1786,33 @@ def __get_spark_session(self):
         if self.spark_init:
             return self.spark
         try:
+            from pyspark.conf import SparkConf
             from pyspark.sql import SparkSession
-            self.spark = SparkSession.builder.master(
-                os.getenv('SPARK_MASTER_HOST', 'local')).getOrCreate()
+
+            repo_config = RepoConfig(repo_path=self.repo_path)
+            spark_config = repo_config.spark_config
+
+            if spark_config:
+                conf = SparkConf()
+                if spark_config.get('app_name'):
+                    conf.setAppName(spark_config.get('app_name'))
+                if spark_config.get('spark_master'):
+                    conf.setMaster(spark_config.get('spark_master'))
+                if spark_config.get('spark_home'):
+                    conf.setSparkHome(spark_config.get('spark_home'))
+                if spark_config.get('spark_jars'):
+                    conf.setJars(spark_config.get('spark_jars'))
```

Comment:
Did you test setJars? It throws error `AttributeError: 'SparkConf' object has no attribute 'setJars'`

Reply:
Good catch: this method is available in Java but not in PySpark, as shown in the Python source code for [SparkConf](https://github.com/apache/spark/blob/master/python/pyspark/conf.py#L171). I found a work-around in [Python SparkConf.setExecutorEnv Examples](https://python.hotexamples.com/examples/pyspark/SparkConf/setExecutorEnv/python-sparkconf-setexecutorenv-method-examples.html), and made a fix. Please review this change. I'll test it out tomorrow as well.